jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/LoggingMXBeanImpl.java

```diff
@@ -23,109 +23,127 @@
 import java.lang.reflect.Method;
 import java.security.AccessController;
 import java.security.PrivilegedAction;
-import java.util.ArrayList;
-import java.util.Enumeration;
-import java.util.List;
-/*[IF !Sidecar19-SE]
+import java.util.Optional;
+/*[ELSE]*/
 import java.util.logging.LoggingMXBean;
 import java.util.logging.Level;
 import java.util.logging.LogManager;
 import java.util.logging.Logger;
 /*[ENDIF]*/
 
 import javax.management.ObjectName;
-import javax.management.StandardMBean;
 
 /**
  * Runtime type for {@link java.lang.management.PlatformLoggingMXBean}.
  *
  * @since 1.5
  */
 public final class LoggingMXBeanImpl
-		extends StandardMBean
 		implements
 /*[IF !Sidecar19-SE]
 			LoggingMXBean,
 /*[ENDIF]*/
 			PlatformLoggingMXBean {
 
-	private static final LoggingMXBeanImpl instance;
-
-/*[IF Sidecar19-SE]*/
-	private static final Method logManager_getLogManager;
-	private static final Method logManager_getLogger;
-	private static final Method logManager_getLoggerNames;
-	private static final Method logger_getLevel;
-	private static final Method logger_getParent;
-	private static final Method logger_getName;
-	private static final Method logger_setLevel;
-	private static final Method level_getName;
-	private static final Method level_parse;
-	private static final String logManager_LOGGING_MXBEAN_NAME;
-/*[ENDIF]*/	
-
-	static {
-/*[IF Sidecar19-SE]*/
-		try {
+	private static final LoggingMXBeanImpl instance = createInstance();
 
-			Module java_logging = ModuleLayer.boot().findModule("java.logging").get(); //$NON-NLS-1$
-			Class<?>[] logClasses = AccessController.doPrivileged((PrivilegedAction<Class<?>[]>)
-				() -> new Class[] {
-					Class.forName(java_logging, "java.util.logging.LogManager"), //$NON-NLS-1$
-					Class.forName(java_logging, "java.util.logging.Logger"), //$NON-NLS-1$
-					Class.forName(java_logging, "java.util.logging.Level") //$NON-NLS-1$
-				});
-			Class<?> logManagerClass = logClasses[0];
-			Class<?> loggerClass = logClasses[1];
-			Class<?> levelClass = logClasses[2];
-			
-			logManager_getLogManager = logManagerClass.getMethod("getLogManager"); //$NON-NLS-1$
-			logManager_getLogger = logManagerClass.getMethod("getLogger", String.class); //$NON-NLS-1$
-			logManager_getLoggerNames = logManagerClass.getMethod("getLoggerNames"); //$NON-NLS-1$
-			logger_getLevel = loggerClass.getMethod("getLevel"); //$NON-NLS-1$
-			logger_getParent = loggerClass.getMethod("getParent"); //$NON-NLS-1$
-			logger_getName = loggerClass.getMethod("getName"); //$NON-NLS-1$
-			logger_setLevel = loggerClass.getMethod("setLevel", levelClass); //$NON-NLS-1$
-			level_getName = levelClass.getMethod("getName"); //$NON-NLS-1$
-			level_parse = levelClass.getMethod("parse", String.class); //$NON-NLS-1$
-			logManager_LOGGING_MXBEAN_NAME = logManagerClass.getField("LOGGING_MXBEAN_NAME").get(null).toString(); //$NON-NLS-1$
+	private static LoggingMXBeanImpl createInstance() {
+		/*[IF Sidecar19-SE]*/
```

Comment:
Can you tag this with 11?  We no longer support 9

Reply:
I used the same flags as were used previously.
If `Java9` was a flag that could be used, it would be more accurate than `Java11`.

Reply:
That's unsustainable.  We should (lazily) garbagecollect unsupported java versions from the JCL preprocessor code.  Ideally, we'll have the 8 & 11 LTS plus the versions prior to the next LTS

Reply:
I would much prefer a scheme that doesn't erase the information about where feature boundaries lie.

What do you think about instead enhancing the preprocessor so we could say:
```
/*[IF JAVA_SPEC_VERSION >= 9]*/
```
mirroring the practise in C:
```
#if JAVA_SPEC_VERSION >= 9
```

Reply:
The preprocessor effectively works that way already.  The `9` is redundant information that no longer provides value.  I'm not sure why we care if a feature was introduced in 9 or 10 as we don't support those versions any more.

Reply:
The preprocessor does not support binary operators besides `&` and `|`.

Reply:
It can't be forgotten, an implementation of the method is required by `PlatformManagedObject`.